v8.0.0-alpha.12 (#16596)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.0.0-alpha.12

_Feb 14, 2025_

We'd like to offer a big thanks to the 16 contributors who made this release possible. Here are some highlights ✨:

- 📦 Data Grid [data source](https://next.mui.com/x/react-data-grid/server-side-data/) is now available in the Community plan

> - ⚡ Improve Data Grid Excel export serialization performance

Reviewer: ```suggestion
- ⚡ Improve Data Grid Excel export serialization performance
- 🚫 Add ["No columns" overlay](https://next.mui.com/x/react-data-grid/overlays/#no-columns-overlay) to Data Grid
```

- 🌍 Improve Polish (pl-PL) and Ukrainian (uk-UA) locales on the Data Grid
- 🐞 Bugfixes

Special thanks go out to the community contributors who have helped make this release possible:

> @Neonin, @nusr and @pawelkula.

Reviewer 2: ```suggestion
@Neonin, @nusr, and @pawelkula.
```

Following are all team members who have contributed to this release:

> @alexfauquette, @arminmeh, @bernardobelchior, @cherniavskii, @Janpot, @JCQuintas, @KenanYusuf, @LukasTy, @MBilalShafi, @michelengelen, @oliviertassinari, @romgrk and @mapache-salvaje.

Reviewer 2: ```suggestion
@alexfauquette, @arminmeh, @bernardobelchior, @cherniavskii, @Janpot, @JCQuintas, @KenanYusuf, @LukasTy, @MBilalShafi, @michelengelen, @oliviertassinari, @romgrk, and @mapache-salvaje.
```

### Data Grid

#### Breaking changes

- The `main--hasSkeletonLoadingOverlay` class has been renamed to `main--hiddenContent` and is now also applied when the "No columns" overlay is displayed.

- The `apiRef.current.forceUpdate()` method was removed. Use selectors combined with `useGridSelector()` hook to react to changes in the state.

> - The selectors signature has been updated. They are only accepting `apiRef` as a first argument.

Reviewer 2: ```suggestion
- The selectors signature has been updated. They are only accepting `apiRef` as a first argument and `instanceId` is no longer the third argument.
```

  ```diff
  -mySelector(state, arguments, instanceId)
  +mySelector(apiRef, arguments)
  ```

#### `@mui/x-data-grid@8.0.0-alpha.12`

- [DataGrid] Add "No columns" overlay (#16543) @KenanYusuf

> - [DataGrid] Add `resetPageOnSortFilter` prop that resets the page after sorting/filtering (#16450) @arminmeh

Reviewer 3: ```suggestion
- [DataGrid] Go to the first page when sorting/filtering is applied (#16447) @arminmeh
```

This one cancels #16450, but the changes from #16450 will stay in v7. Since it is all in one release, I think that just mentioning new behavior in the release notes is enough

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.0.0-alpha.12

_Feb 14, 2025_

We'd like to offer a big thanks to the 16 contributors who made this release possible. Here are some highlights ✨:

> 

Reviewer 3: ```suggestion

📦 Data Grid [data source](https://next.mui.com/x/react-data-grid/server-side-data/) is now available in the Community plan
```

- 🌍 Improve Polish (pl-PL) and Ukrainian (uk-UA) locales on the Data Grid
- 🐞 Bugfixes

Special thanks go out to the community contributors who have helped make this release possible:

> @mapache-salvaje, @Neonin, @nusr, @pawelkula, .

Reviewer 3: ```suggestion
@Neonin, @nusr and @pawelkula.
```

Following are all team members who have contributed to this release:

> @alexfauquette, @arminmeh, @bernardobelchior, @cherniavskii, @Janpot, @JCQuintas, @KenanYusuf, @LukasTy, @MBilalShafi, @michelengelen, @oliviertassinari and @romgrk.

Reviewer 3: ```suggestion
@alexfauquette, @arminmeh, @bernardobelchior, @cherniavskii, @Janpot, @JCQuintas, @KenanYusuf, @LukasTy, @MBilalShafi, @michelengelen, @oliviertassinari, @romgrk and @mapache-salvaje.
```

### Data Grid

#### Breaking changes

- The `main--hasSkeletonLoadingOverlay` class has been renamed to `main--hiddenContent` and is now also applied when the "No columns" overlay is displayed.

- The `apiRef.current.forceUpdate()` method was removed. Use selectors combined with `useGridSelector()` hook to react to changes in the state.

> - Fixes #16220
> 
>   - Before: https://codesandbox.io/p/sandbox/intelligent-leavitt-q4h68n?from-embed=&workspaceId=ws_96AUJAP4MpFUq5gdSxRein
>   - After: https://codesandbox.io/p/sandbox/stupefied-scott-qnqpx5
> 

Reviewer 4: It looks like these lines do not belong here. 🙈 
```suggestion
```

Author: @MBilalShafi it you can rework your changelog breaking change to be more comprehensible :pray:

Author: Removed

> - Fixes #16220
> 
>   - Before: https://codesandbox.io/p/sandbox/intelligent-leavitt-q4h68n?from-embed=&workspaceId=ws_96AUJAP4MpFUq5gdSxRein
>   - After: https://codesandbox.io/p/sandbox/stupefied-scott-qnqpx5
> 
> - Add a skeleton loading overlay on the aggregation row while it's loading
> 
>   https://github.com/user-attachments/assets/310f57fe-2809-45c7-9930-22f5df413e04
> 

Reviewer 3: ```suggestion
```

Author: You would remove the two highlight altogether?

Reviewer 2: +1 on removing from the breaking changes section

- The selectors signature has been updated. They are only accepting `apiRef` as a first argument.

  ```diff
  -mySelector(state, arguments, instanceId)
  +mySelector(apiRef, arguments)
  ```

#### `@mui/x-data-grid@v8.0.0-alpha.12`

- [DataGrid] Add "No columns" overlay (#16543) @KenanYusuf
- [DataGrid] Add `resetPageOnSortFilter` prop that resets the page after sorting/filtering (#16450) @arminmeh
- [DataGrid] All selectors accept only `apiRef` as first argument (#16198) @arminmeh
- [DataGrid] Avoid `undefined` value for pagination `rowCount` (#16488) @LukasTy
- [DataGrid] Create the base Checkbox slot (#16445) @romgrk
- [DataGrid] Create the base Input slot (#16443) @romgrk
- [DataGrid] Create the base MenuList slot (#16481) @romgrk
- [DataGrid] Create the base Popper slot (#16362) @romgrk
- [DataGrid] Create the base Select slot (#16394) @romgrk
- [DataGrid] Create the base Switch slot (#16527) @romgrk

> - [DataGrid] Extract `getRowId` API method as a selector (#16487) @MBilalShafi

Reviewer 3: ```suggestion
- [DataGrid] Extract `getRowId()` API method as a selector (#16487) @MBilalShafi
```

- [DataGrid] Fix the `onClock` prop of the base Select slot (#16557) @romgrk
- [DataGrid] Make base data source available in the Community plan (#16359) @MBilalShafi
- [DataGrid] Remove `apiRef.current.forceUpdate()` method (#16560) @MBilalShafi

> - [DataGrid] Fix the unexpected behavior of the pagination when using -1 for "All" rows per page (#16485) @nusr

Reviewer 3: ```suggestion
- [DataGrid] Fix the unexpected behavior of the pagination when using `-1` for "All" rows per page (#16485) @nusr
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.0.0-alpha.12

_Feb 14, 2025_

We'd like to offer a big thanks to the 16 contributors who made this release possible. Here are some highlights ✨:

> 📦 Data Grid [data source](https://next.mui.com/x/react-data-grid/server-side-data/) is now available in the Community plan

Reviewer 5: ```suggestion
- 📦 Data Grid [data source](https://next.mui.com/x/react-data-grid/server-side-data/) is now available in the Community plan
```

> - 🌍 Improve Polish (pl-PL) and Ukrainian (uk-UA) locales on the Data Grid

Reviewer 5: ```suggestion
- ⚡ Improve Data Grid Excel export serialization performance
- 🌍 Improve Polish (pl-PL) and Ukrainian (uk-UA) locales on the Data Grid
```
